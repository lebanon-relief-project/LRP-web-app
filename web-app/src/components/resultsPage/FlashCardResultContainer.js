import React, { useState } from "react";
import styled from "styled-components";
import devices from "../../styles/Devices";
import { ResultCard } from "./ResultCard";
import { getCardTitleFromSessionStorage } from "../../util/util";
import Colours from "../../styles/Colours";

// @param results: ResultsResponse
export const FlashCardResultContainer = ({ results }) => {
  const resultsResponseArray = results.results;

  // Setup state to display first flashcards recommendations
  const setInitialState = (resultsResponseArray) => {
    let initialState = [];

    resultsResponseArray.forEach((result, index) => {
      if (index === 0) {
        initialState.push({ id: result._id, selected: true });
      } else {
        initialState.push({ id: result._id, selected: false });
      }
    });

    return initialState;
  };

  const [selectedFlashCardState, setSelectedFlashCardState] = useState(
    setInitialState(resultsResponseArray)
  );

  // If a new flashcard is clicked update state to render that cards recommendations
  const updateSelectedFlashCardState = (id) => {
    let updatedState = [];

    selectedFlashCardState.forEach((result) => {
      if (result.id === id) {
        updatedState.push({ id: result.id, selected: true });
      } else {
        updatedState.push({ id: result.id, selected: false });
      }
    });

    setSelectedFlashCardState(updatedState);
  };

  // If card is selected make title bold
  const renderTitles = () => {
    return selectedFlashCardState.map((card, index) => {
      if (card.selected) {
        return (
          <SelectedStyledFlashCardTitle
            data-testid={`flashcardTitleButton_${index}`}
            key={`flashcardTitleButton_${index}`}
            onClick={() => updateSelectedFlashCardState(card.id)}
          >
            {getCardTitleFromSessionStorage(card.id)}
          </SelectedStyledFlashCardTitle>
        );
      } else {
        return (
          <UnSelectedStyledFlashCardTitle
            data-testid={`flashcardTitleButton_${index}`}
            key={`flashcardTitleButton_${index}`}
            onClick={() => updateSelectedFlashCardState(card.id)}
          >
            {getCardTitleFromSessionStorage(card.id)}
          </UnSelectedStyledFlashCardTitle>
        );
      }
    });
  };

  // Only render the selected cards recommendations and image
  const renderRecommendations = () => {
    return selectedFlashCardState
      .filter((card) => card.selected)
      .map((card) => {
        const selectedFlashcardResult = resultsResponseArray.filter(
          (result) => result._id === card.id
        );
        return (
          <ResultCardsWrapper key={`resultCards_wrapper`}>
            <ResultCardImage src={selectedFlashcardResult[0].image} />
            <H1>{selectedFlashcardResult[0].expl_title}</H1>
            <P>{selectedFlashcardResult[0].expl_body}</P>
            {selectedFlashcardResult[0].recommendations.map(
              (recommendation, index) => {
                return (
                  <ResultCard
                    title={recommendation.title}
                    body={recommendation.body}
                    key={`flashcard_result_${index}`}
                    data-testid={`${recommendation.title}_flashcard_result_${index}`}
                  />
                );
              }
            )}
          </ResultCardsWrapper>
        );
      });
  };

  return (
    <Container>
      <TitlesContainer>
        <Background />
        <TitlesForeground>{renderTitles()}</TitlesForeground>
      </TitlesContainer>
      {renderRecommendations()}
    </Container>
  );
};

const H1 = styled.h1`
  margin-left: 10px;
  font-weight: bold;

  @media (max-width: ${devices.mobile}) {
    margin: 18px;
  }
`;
const P = styled.p`
  margin: 10px;
  font-size: 16px;

  @media (max-width: ${devices.mobile}) {
    margin: 18px;
  }
`;

const TitlesForeground = styled.div`
  display: flex;
  width: auto;
  height: 100%;
  z-index: 1;
  position: absolute;
  flex-direction: column;
  overflow: auto;

  @media (max-width: ${devices.ipadpro}) {
    flex-direction: row;
    padding: 0 16px;
    white-space: nowrap;
  }
`;

const Background = styled.div`
  display: flex;
  width: 100%;
  height: 100%;
  z-index: 0;
  border-left: 6px solid rgba(255, 197, 61, 0.2);
  position: absolute;

  @media (max-width: ${devices.ipadpro}) {
    border: none;
    width: 1000px;
    border-top: 4px solid rgba(255, 197, 61, 0.2);
  }
`;

const TitlesContainer = styled.h1`
  width: 100%;
  max-width: 270px;
  height: 100%;
  position: relative;
  display: flex;
  @media (max-width: ${devices.ipadpro}) {
    max-width: 100%;
    width: auto;
    height: 78px;
    overflow: auto;
  }
`;

const Container = styled.div`
  display: flex;
  flex: 1;
  @media (max-width: ${devices.ipadpro}) {
    flex-direction: column;
  }
`;

const ResultCardsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  @media (max-width: ${devices.ipadpro}) {
    max-width: 100%;
  }
`;

const ResultCardImage = styled.img`
  height: 100px;
  width: 244px;
  margin: 10px;

  @media (max-width: ${devices.mobile}) {
    margin: 18px;
  }
`;

const SelectedStyledFlashCardTitle = styled.button`
  color: ${Colours.darkBlue};
  font-weight: bold;
  font-size: 24px;
  min-width: 270px;
  text-align: left;
  padding-left: 24px;
  padding-top: 30px;
  padding-bottom: 30px;
  background: none;
  border: none;
  border-left: solid;
  border-left-color: ${Colours.gold};
  border-left-width: 6px;
  cursor: pointer;
  outline: inherit;
  @media (max-width: ${devices.ipadpro}) {
    border: none;
    border-top: 4px solid ${Colours.gold};
    max-width: 100%;
    padding: 0;
    min-width: auto;
    margin-right: 30px;
  }
`;
const UnSelectedStyledFlashCardTitle = styled(SelectedStyledFlashCardTitle)`
  opacity: 0.3;
  border-left-color: transparent;
  @media (max-width: ${devices.ipadpro}) {
    opacity: 0.3;
    border-top: 4px solid transparent;
    max-width: 100%;
    padding: 0;
  }
`;

export default FlashCardResultContainer;
