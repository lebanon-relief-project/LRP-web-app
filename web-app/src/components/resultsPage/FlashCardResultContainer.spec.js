import React from "react";
import { render } from "@testing-library/react";
import FlashCardResultContainer from "./FlashCardResultContainer";
import "jest-styled-components";

describe("the FlashCardResultContainer component", () => {
  let container;
  let results = {
    results: [
      {
        _id: "test ID",
        expl_title: "test",
        expl_body: "test",
        image: "test",
        recommendations: [{ title: "test", body: "test" }],
      },
    ],
  };

  it("should match the snapshot", () => {
    ({ container } = render(<FlashCardResultContainer results={results} />));

    expect(container).toMatchSnapshot();
  });

  it.todo("should render the selected flashcards recommendations only");
});
