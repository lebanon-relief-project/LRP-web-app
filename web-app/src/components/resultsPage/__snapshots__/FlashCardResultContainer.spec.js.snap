// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the FlashCardResultContainer component should match the snapshot 1`] = `
.c10 {
  box-shadow: 0 0px 8px 0 rgba(0,0,0,0.2);
  -webkit-transition: 0.3s;
  transition: 0.3s;
  Offset: 0px,1px rgba(0,0,0,0.25) border-radius:10px;
  padding: 6px;
  margin: 10px;
}

.c8 {
  margin-left: 10px;
  font-weight: bold;
}

.c9 {
  margin: 10px;
  font-size: 16px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: auto;
  height: 100%;
  z-index: 1;
  position: absolute;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  z-index: 0;
  border-left: 6px solid rgba(255,197,61,0.2);
  position: absolute;
}

.c1 {
  width: 100%;
  max-width: 270px;
  height: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.c7 {
  height: 100px;
  width: 244px;
  margin: 10px;
}

.c4 {
  color: #002766;
  font-weight: bold;
  font-size: 24px;
  min-width: 270px;
  text-align: left;
  padding-left: 24px;
  padding-top: 30px;
  padding-bottom: 30px;
  background: none;
  border: none;
  border-left: solid;
  border-left-color: #FFC53D;
  border-left-width: 6px;
  cursor: pointer;
  outline: inherit;
}

.c5 {
  opacity: 0.3;
  border-left-color: transparent;
}

@media (max-width:576px) {
  .c8 {
    margin: 18px;
  }
}

@media (max-width:576px) {
  .c9 {
    margin: 18px;
  }
}

@media (max-width:1025px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 0 16px;
    white-space: nowrap;
  }
}

@media (max-width:1025px) {
  .c2 {
    border: none;
    width: 1000px;
    border-top: 4px solid rgba(255,197,61,0.2);
  }
}

@media (max-width:1025px) {
  .c1 {
    max-width: 100%;
    width: auto;
    height: 78px;
    overflow: auto;
  }
}

@media (max-width:1025px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:1025px) {
  .c6 {
    max-width: 100%;
  }
}

@media (max-width:576px) {
  .c7 {
    margin: 18px;
  }
}

@media (max-width:1025px) {
  .c4 {
    border: none;
    border-top: 4px solid #FFC53D;
    max-width: 100%;
    padding: 0;
    min-width: auto;
    margin-right: 30px;
  }
}

@media (max-width:1025px) {
  .c5 {
    opacity: 0.3;
    border-top: 4px solid transparent;
    max-width: 100%;
    padding: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      <div
        class="c2"
      />
      <div
        class="c3"
      >
        <button
          class="c4"
          data-testid="flashcardTitleButton_0"
        >
          test title
        </button>
        <button
          class="c4 c5"
          data-testid="flashcardTitleButton_1"
        >
          test title
        </button>
        <button
          class="c4 c5"
          data-testid="flashcardTitleButton_2"
        >
          test title
        </button>
      </div>
    </h1>
    <div
      class="c6"
    >
      <img
        class="c7"
        src="feeling-guilty.svg"
      />
      <h1
        class="c8"
      >
        Feeling Guilty
      </h1>
      <p
        class="c9"
      />
      <div
        class="c10"
      >
        <div
          id="result card"
        >
          <h4
            style="margin: 10px; font-size: 20px;"
          >
            <b>
              Allow yourself to feel
            </b>
          </h4>
          <p
            style="margin: 10px; font-size: 16px;"
          >
            It can be helpful for you to accept the feelings that surface by allowing yourself time to process the guilt. This could be achieved by accepting that you are not in control of the event; acknowledging  how you reacted at the time of the event; and understanding that you acted to the best of your ability.
          </p>
        </div>
      </div>
      <div
        class="c10"
      >
        <div
          id="result card"
        >
          <h4
            style="margin: 10px; font-size: 20px;"
          >
            <b>
              Talk about it
            </b>
          </h4>
          <p
            style="margin: 10px; font-size: 16px;"
          >
            Share your experience, thoughts, and feelings. It's important that you only do so when you feel ready to. This can be a gradual process. Take your time and do it at whatever pace feels comfortable for you.
          </p>
        </div>
      </div>
      <div
        class="c10"
      >
        <div
          id="result card"
        >
          <h4
            style="margin: 10px; font-size: 20px;"
          >
            <b>
              Try these excercises
            </b>
          </h4>
          <p
            style="margin: 10px; font-size: 16px;"
          >
            Lorem ipsum...
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
